src/palace/manager/celery/tasks/search.py

```diff
@@ -113,34 +122,46 @@ def update_read_pointer(task: Task) -> None:
     )
 
 
+@shared_task(queue=QueueNames.default, bind=True)
+def search_indexing(task: Task, batch_size: int = 500) -> None:
+    redis_client = task.services.redis.client()
+    task_lock = TaskLock(redis_client, task)
+    with task_lock.lock() as acquired:
+        if not acquired:
+            raise BasePalaceException(f"{task.name} is already running.")
+
+        waiting = WaitingForIndexing(redis_client)
+        works = waiting.pop(batch_size)
+
+        if len(works) > 0:
+            index_works.delay(works=works)
+
+        if len(works) == batch_size:
```

Comment:
What you suggest here is true, it can happen that the set contains the batch size of works, so we will queue another task. However its not a problem if that happens. 

On the next iteration the task will try to pop more works, and find an empty set, then exit. We have to handle this case anyway, where the task gets run with an empty set, because its possible that there are no works waiting for indexing when the task is scheduled to be run.